### PR TITLE
[GR-1111] cfMeDIP report 

### PR DESCRIPTION
**Pull request by avarsava**
Still don't know whether the % PF Reads Aligned should be multiplied by 100 or not

### CODE REVIEW

File: application/dash_application/views/single_lane_cfmedip.py [hunk 1]
```diff
@@ -0,0 +1,628 @@
...
+# Add shape, colour, and size cols to dataframe 
+cfmedip = add_graphable_cols(cfmedip, initial, shape_colour.items_for_df())
+
+
+def generate_number_windows(current_data, graph_params):
+    return generate(
+        "Log10 # Windows at 1, 10, 50, 100x",
+        current_data,
+        lambda d: d[PINERY_COL.SampleName],
+        [lambda d: d[CFMEDIP_COL.NumWindowsWith1Reads],
```
Comment (apmasell) [inline application/dash_application/views/single_lane_cfmedip.py]: Can you indent this a bit more clearly?

**Review by apmasell — APPROVED** [review]

File: application/dash_application/views/single_lane_cfmedip.py (hunk 1 above)
Comment (avarsava) [reply application/dash_application/views/single_lane_cfmedip.py]: Done!

**Comment by slazicoicr** [comment]
I get a crash with:
* All Projects
* Highlight sample: AIX_0008_Ct_T_PE_160101_CM
* Show Data Labels: All

```
Traceback (most recent call last):
  File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/dash/dash.py", line 967, in dispatch
    response.set_data(func(*args, outputs_list=outputs_list))
  File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/dash/dash.py", line 902, in add_context
    output_value = func(*args, **kwargs)  # %% callback invoked %%
  File "/home/slazic/Documents/Code/2019-12-10-Dashi/application/dash_application/views/single_lane_cfmedip.py", line 528, in update_pressed
    generate_number_windows(df, graph_params),
  File "/home/slazic/Documents/Code/2019-12-10-Dashi/application/dash_application/views/single_lane_cfmedip.py", line 165, in generate_number_windows
    graph_params["shownames_val"]
  File "/home/slazic/Documents/Code/2019-12-10-Dashi/application/dash_application/utility/plot_builder.py", line 322, in generate
    y=y_fn(highlight_df),
TypeError: 'list' object is not callable
```

**Comment by slazicoicr** [comment]
Regarding % PF Reads Aligned. Just looking at the data, I think it needs to be multiplied by 100%, but I have not worked on this workflow.

**Comment by avarsava** [comment]
> Regarding % PF Reads Aligned. Just looking at the data, I think it needs to be multiplied by 100%, but I have not worked on this workflow.

Yeah I asked on Slack and got an inconclusive answer there too :(

**Comment by avarsava** [comment]
> I get a crash with:
> 
>     * All Projects
> 
>     * Highlight sample: AIX_0008_Ct_T_PE_160101_CM
> 
>     * Show Data Labels: All
> 
> 
> ```
> Traceback (most recent call last):
>   File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
>     response = self.full_dispatch_request()
>   File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
>     rv = self.handle_user_exception(e)
>   File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
>     reraise(exc_type, exc_value, tb)
>   File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
>     raise value
>   File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
>     rv = self.dispatch_request()
>   File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
>     return self.view_functions[rule.endpoint](**req.view_args)
>   File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/dash/dash.py", line 967, in dispatch
>     response.set_data(func(*args, outputs_list=outputs_list))
>   File "/home/slazic/.local/miniconda3/envs/platonic_dash/lib/python3.6/site-packages/dash/dash.py", line 902, in add_context
>     output_value = func(*args, **kwargs)  # %% callback invoked %%
>   File "/home/slazic/Documents/Code/2019-12-10-Dashi/application/dash_application/views/single_lane_cfmedip.py", line 528, in update_pressed
>     generate_number_windows(df, graph_params),
>   File "/home/slazic/Documents/Code/2019-12-10-Dashi/application/dash_application/views/single_lane_cfmedip.py", line 165, in generate_number_windows
>     graph_params["shownames_val"]
>   File "/home/slazic/Documents/Code/2019-12-10-Dashi/application/dash_application/utility/plot_builder.py", line 322, in generate
>     y=y_fn(highlight_df),
> TypeError: 'list' object is not callable
> ```

Fixed

**Comment by slazicoicr** [comment]
Go ahead and multiply % PF Reads Aligned by 100, because right now that graph looks very off.

File: application/dash_application/utility/plot_builder.py [hunk 2]
```diff
@@ -344,20 +316,37 @@ def generate(title_text, sorted_data, x_fn, y_fn, axis_text, colourby, shapeby,
...
-            showlegend=False,
-            marker={
-                "symbol": highlight_df['shape'],
-                "color": highlight_df['colour'],
-                "size": highlight_df['markersize'],
-                "opacity": 1
-            }
-        ))
+        if isinstance(y_fn, list):
+            for fn in y_fn: # Don't like looping over this twice but unsure whether these can be guaranteed to be in foreground otherwise
```
Comment (slazicoicr) [inline application/dash_application/utility/plot_builder.py:320]: If you don't like the second for loop, have the first for loop append highlight traces to a new list (`highlight_traces`).  Once the loop finishes `traces =  traces + highlight_traces` should ensure highlight is always on top.

**Review by slazicoicr — APPROVED** [review]

**Pull request merged**